app/Commands/CheckDockerComposeVersion.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Commands;
+
+use App\CommandExecutor;
+use App\Builder\DockerCompose;
+use App\Commands\Traits\RunTask;
+use LaravelZero\Framework\Commands\Command;
+
+class CheckDockerComposeVersion extends Command
+{
+    const COMPOSE_MIN_VERSION = '1.24';
+
+    use RunTask;
+
+    /**
+     * The signature of the command.
+     *
+     * @var string
+     */
+    protected $signature = 'check-docker-compose-version';
+
+    /**
+     * The description of the command.
+     *
+     * @var string
+     */
+    protected $description = 'Checks that the docker-compose version is compatible';
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle(CommandExecutor $executor)
+    {
+        return $this->runTask('Checking docker-compose version', function () use ($executor) {
+            return $this->checkDockerComposeVersion($executor);
+        });
+    }
+
+    protected function checkDockerComposeVersion(CommandExecutor $executor): int
+    {
+        if ($executor->runQuietly(new DockerCompose('-v'))) {
+            return 1;
+        }
+
+        $output = $executor->getOutputBuffer();
+        $matches = [];
+
+        if (! preg_match('/(?:(\d+)\.)(?:(\d+)\.)?(\*|\d+)/', $output, $matches)) {
```

Comment:
maybe we could use https://github.com/composer/semver

Reply:
or other lib for doing the version matching

Reply:
oh you're just cleaning docker output, nvm i guess

Reply:
this piece is actually extracting the version from the `docker-compose -v`, so the regex is probably inevitable.. I guess you meant to use semver instead of `version_compare` - but that one is simple enough it seems.

Reply:
you can use `docker-compose version --short`

Reply:
ah, scratch this.. just saw the other note on formatting the output of `-v`.